econml/dml/dml.py

```diff
@@ -29,74 +29,85 @@
 from ..sklearn_extensions.model_selection import WeightedStratifiedKFold
 from ..utilities import (_deprecate_positional, add_intercept,
                          broadcast_unit_treatments, check_high_dimensional,
-                         cross_product, deprecated, fit_with_groups,
+                         cross_product, deprecated,
                          hstack, inverse_onehot, ndim, reshape,
                          reshape_treatmentwise_effects, shape, transpose,
                          get_feature_names_or_default, filter_none_kwargs)
 from .._shap import _shap_explain_model_cate
+from ..sklearn_extensions.model_selection import get_selector, ModelSelector, SingleModelSelector
 
 
-class _FirstStageWrapper:
-    def __init__(self, model, is_Y, featurizer, linear_first_stages, discrete_treatment):
```

Comment:
Is linear_first_stages used anymore?

Reply:
No; linear_first_stages has always been an awkward hack and this gives us a good opportunity to get rid of it.  Given

T = f(X, W) + error
Y = T <alpha phi(X)> + g(X,W) + error

if both f and g are linear, then because we multiply T by theta in the equation for Y, Y is not actually a linear function of X and W, we need to include interaction terms.  So we introduced linear_first_stages because our default y and t models were linear and we wanted to make sure we could estimate the correct model if f and g were linear.  Now, since we do model selection, the y and t models are not (necessarily) linear, so this makes less sense.

There's also a potentially big performance cost if users accidentally leave linear_first_stages set to True, because we can end up generating a huge number of interaction terms, and we see this happen fairly often in practice.

Additionally, linear models aren't the only time that specifying the right models for f and g as your t- and y-models wouldn't give the correct estimate - if f and g are both degree-n polynomials then you'd need to interact the degree-n interaction terms of [X;W] with the featurized X, for example, and we've never supported that; users can always supply a more complicated pipelined model that does something like that themselves if they'd like, but I'd argue it doesn't make sense to try to solve it on our end.